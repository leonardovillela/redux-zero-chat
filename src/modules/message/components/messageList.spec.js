import React from 'react';
import { shallow } from 'enzyme';

import MessageList from './messageList';

describe('Test MessageList component unit', () => {
  it('Should render component with correct structure', () => {
    const messagesStub1 = [];
    const wrapper1 = shallow(<MessageList className="Teste 1" messages={messagesStub1} />);

    expect(wrapper1).toMatchSnapshot();

    const messagesStub2 = [{}];
    const wrapper2 = shallow(<MessageList className="Teste 2" messages={messagesStub2} />);

    expect(wrapper2).toMatchSnapshot();

    const messagesStub3 = [{}, {}, {}];
    const wrapper3 = shallow(<MessageList className="Teste 3" messages={messagesStub3} />);

    expect(wrapper3).toMatchSnapshot();
  })

  it('Should render component with correct with correct structure, even when messages undefined', () => {
    const wrapper = shallow(<MessageList />);

    expect(wrapper).toMatchSnapshot();
  });
});
