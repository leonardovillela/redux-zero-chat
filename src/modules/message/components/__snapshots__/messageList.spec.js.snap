// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MessageList component unit Should render component with correct structure 1`] = `
<div
  className="Teste 1 ba pa3 overflow-y-auto"
/>
`;

exports[`Test MessageList component unit Should render component with correct structure 2`] = `
<div
  className="Teste 2 ba pa3 overflow-y-auto"
>
  <MessageListItem
    key="message_item_0"
  />
</div>
`;

exports[`Test MessageList component unit Should render component with correct structure 3`] = `
<div
  className="Teste 3 ba pa3 overflow-y-auto"
>
  <MessageListItem
    key="message_item_0"
  />
  <MessageListItem
    key="message_item_1"
  />
  <MessageListItem
    key="message_item_2"
  />
</div>
`;

exports[`Test MessageList component unit Should render component with correct with correct structure, even when messages undefined 1`] = `
<div
  className="undefined ba pa3 overflow-y-auto"
/>
`;
